winml/lib/Api.Image/TensorToVideoFrameConverter.cpp

```diff
@@ -345,7 +381,13 @@ void TensorToVideoFrameConverter::ConvertGPUTensorToDX12Texture(
   CD3DX12_RECT scissorRect(0, 0, (LONG)outputDesc.Width, outputDesc.Height);
   ComPtr<ID3D12Device> spDx12Device = device_cache.GetD3D12Device();
 
-  GPUTensorToDX12TextureTelemetryEvent telemetrylogger(tensorDesc);
+
+  std::unique_ptr<GPUTensorToDX12TextureTelemetryEvent> telemetryLogger;
+  // we're inside a lock from the caller of this function, so it's ok to use this static
+  static EventTimer eventTimer;
+  if (eventTimer.Start()) {
+    telemetryLogger = std::make_unique<GPUTensorToDX12TextureTelemetryEvent>(tensorDesc);
+  }
```

Comment:
Do we need the heap allocation? Doesn't `std::optional` suffice?

Reply:
I didn't know about std::optional! I will take a look, thanks!